avalon/components/avalon_component.py

```diff
@@ -91,10 +106,68 @@ def handle_avalon_push_artifact(self, event, *args, **kwargs):
 
         return self.actions.push_single_resilient_artifact(incident, artifact)
 
+    # Handles "Avalon: Start Auto-refresh" action.
+    @handler("avalon_start_auto_refresh")
+    def handle_avalon_start_auto_refresh_workflow(self, event, *args, **kwargs):
+        incident = event.message["incident"]
+        logger.info("Called from incident {}: {}".format(incident["id"], incident["name"]))
+
+        incident_id = incident["id"]
+
+        self._start_auto_refresh(incident)
+
+        # set the auto refresh field to false
+        old_value = self.actions.res.incident_get_avalon_auto_refresh(incident) 
+        self.actions.res.incident_set_avalon_auto_refresh(incident_id, True, old_value)
+
+        return "Avalon Auto-refresh started successfully."
+
+
     # Handles "Avalon: Stop Auto-refresh Workflow" action.
-    @handler("avalon_stop_auto_refresh_workflow")
-    def handle_avalon_stop_auto_refresh_workflow(self, event, *args, **kwargs):
+    @handler("avalon_stop_auto_refresh")
+    def handle_avalon_stop_auto_refresh(self, event, *args, **kwargs):
         incident = event.message["incident"]
         logger.info("Called from incident {}: {}".format(incident["id"], incident["name"]))
 
-        return self.actions.stop_auto_refresh_workflow(incident)
+        # remove timer the incident map
+        incident_id = incident["id"]
+        timer = self.incident_timer.pop(incident_id, None)
+        
+        # stop ticking
+        if timer:
+            timer.unregister()    
+
+        # set the auto refresh field to false
+        old_value = self.actions.res.incident_get_avalon_auto_refresh(incident) 
+        self.actions.res.incident_set_avalon_auto_refresh(incident_id, False, old_value)
+
+        return "Avalon Auto-refresh stopped successfully."
+
+    # Handles the auto refresh timer events
+    @handler("auto_refresh")
+    def handle_auto_refresh(self, event, *args, **kwargs):
+        incident_id = event.incident_id
+        incident = self.actions.res.incident_get(incident_id)
+
+        logger.info("handle_auto_refresh called for incident {}: {}".format(incident["id"], incident["name"]))
+
+        self.actions.refresh_nodes(incident)
+
+
+    def _start_auto_refresh(self, incident):
+        # Get the interval from the avalon_auto_refresh_time field 
+        refresh_interval_minutes = self.actions.res.incident_get_avalon_auto_refresh_time(incident)
+
+        # create timer
+        incident_id = incident["id"]
+        auto_refresh_event = auto_refresh(incident_id)
+        timer = Timer(60 * refresh_interval_minutes, auto_refresh_event, self.channel, persist=True)
```

Comment:
May need to check that `refresh_interval_minutes` is not None. I'm getting that error on my test instance.

Reply:
Now defaults to 60 minutes if not set in IBM Resilient